### PR TITLE
Brooklyne.finni/alert shortcode [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by brooklynebkbk**
### What does this PR do?

- creates a reusable shortcode for alerts that can be included in md
- replaces instances of html syntax for alerts in md files with new shortcode

### Motivation

https://datadoghq.atlassian.net/jira/software/projects/WEB/boards/102?assignee=5fd13b2734847e0069ea3ec5&selectedIssue=WEB-1617

### Preview

- many many pages throughout site
- the site should look unchnaged

preview branch:
https://docs-staging.datadoghq.com/brooklyne.finni/alert_shortcode/

example pages: 
https://docs-staging.datadoghq.com/brooklyne.finni/alert_shortcode/tracing/setup_overview/setup/dotnet-core/?tab=windows

https://docs-staging.datadoghq.com/brooklyne.finni/alert_shortcode/events/

https://docs-staging.datadoghq.com/brooklyne.finni/alert_shortcode/fr/tracing/setup_overview/setup/dotnet-core/?tab=windows

https://docs-staging.datadoghq.com/brooklyne.finni/alert_shortcode/ja/dashboards/

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by davidejones** [comment]
@brooklynebkbk I still see a number of markdown files that could have changes try running `grep -r --color "<div class=\"alert" ./content/en` to see which.

File: layouts/shortcodes/alert.html [hunk 1]
```diff
@@ -0,0 +1 @@
+<div class="alert {{ with .Get `type` }}alert-{{ . }}{{ else }}alert-warning{{ end }}">{{ .Inner }}</div>
```
Comment (davidejones) [inline layouts/shortcodes/alert.html:1]: adding dashes to inner might be a good idea to reduce whitespace `{{- .Inner -}}` or we could also wrap it with something like ```{{- with .Inner -}}...yourexistingcode...{{- end -}}``` so it reduces whitespace and if you did `{{< alert >}}{{< alert />}}` nothing would be output.

File: layouts/api/list.html [hunk 2]
```diff
@@ -126,9 +126,9 @@ <h2 class="mb-2" id="{{ $anchorStr | anchorize }}">
                     <!-- if this is deprecated and we have a higher version link to that -->
                     {{ if (and $endpointVisibility.isVisibleVersion (eq $versionNum "v1") (gt $versionCount 1) (eq (index $tag "x-deprecated") true)) }}}}
                       {{ $collapsedAnchorStr := (print $endpoint.action.summary " " (replace $versionNum "1" "2")) }}
-                      <div class="alert alert-danger">
+                      {{< alert type="danger" >}}
                         <p class="alert-warning">This endpoint is deprecated use the latest endpoint here <a class="toggle-version-tab" href="#{{ $collapsedAnchorStr | anchorize }}">{{- $collapsedAnchorStr -}}.</a></p>
-                      </div>
+                      {{< /alert >}}
```
Comment (davidejones) [inline layouts/api/list.html:131]: I think this is md shortcode syntax in a html file. I get errors when trying to run this locally

File: layouts/shortcodes/alert.html (hunk 1 above)
Comment (lucperkins) [reply layouts/shortcodes/alert.html:1]: I'd also recommend parsing the content as Markdown so that the residual raw HTML is no longer necessary

**Comment by apigirl** [comment]
Closing in favor of the new design after discussion with Booklyne and David

**Pull request closed without merging**